client/client.go

```diff
@@ -173,8 +179,8 @@ func (c client) StartProvider(hostID string, providerRef string, linkName string
 }
 
 // NATs topic: linkdefs.put
-func (c client) AdvertiseLink(actorID string, providerID string, contractID string, linkName string, values map[string]string) []string {
-	subject := broker.AdvertiseLink(c.nsprefix)
+func (c Client) AdvertiseLink(actorID string, providerID string, contractID string, linkName string, values map[string]string) []string {
+	subject := broker.AdvertiseLink(c.nsPrefix)
```

Comment:
As of the yet-to-be-released v0.60 of the OTP host, link management is done by querying the KV bucket directly. The host still responds to messages related to links, but this functionality will be deprecated and eventually removed

To preserve backwards compatibility, v0.23 of the rust control interface client checks if the bucket exists in its constructor/builder, to determine if it should send messages directly to NATS or still publish to the host. Consider following that pattern here

EDIT: I noticed you're calling `kv.GetKVStore(cb.nc, cb.nsPrefix, cb.jsDomain)` in `Build` and quitting if the bucket doesn't exist. If this client is only meant to be compatible with v0.60+ versions of the OTP host, then these link functions should interact with the bucket directly, i.e. call your functions like `PutLink` you added in `kv.go`

Reply:
Good call. I'll dig into this

Reply:
Looking into this, there are some structs I need to create for the Client commands.  right now they are all returning []string where they should be returning the correct acks and what not.  

I will need to look at the rust code and make sure I have the returns for all this sorted.

Reply:
this _should_ be sorted now???